src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -115,12 +118,12 @@ public static string GetFileName(string path, bool includeExtension = true)
         /// Unlike <see cref="System.IO.Path.GetDirectoryName(string)"/> it doesn't check for invalid path characters
         /// </remarks>
         /// <returns>Prefix of path that represents a directory</returns>
-        public static string GetDirectoryName(string path)
+        public static string? GetDirectoryName(string? path)
         {
             return GetDirectoryName(path, IsUnixLikePlatform);
         }
 
-        private static string GetDirectoryName(string path, bool isUnixLike)
+        private static string? GetDirectoryName(string? path, bool isUnixLike)
```

Comment:
Should be `NotNullIfNotNull("path")`, and above.

Reply:
That would make things so much easier, but I don't think that's possible. 
I see a path where `path` is not null, but we still return null from the method

---
In reply to: [377802189](https://github.com/dotnet/roslyn/pull/41063#discussion_r377802189) [](ancestors = 377802189)